### ISSUE TITLE
Revisit maximizing

### ISSUE THREAD

**Issue by ClementTsang** [ClementTsang/bottom#19]
## Describe the feature request

Maximizing is currently fine in that it literally just does that - maximize.

Now, time to take a look at part 2 - adding more details if possible.  Stuff that can't fit on the normal size.

## Additional context

Provide any additional context here.

**Comment by ClementTsang** [comment]
We can also (I guess) use this to extend some other widgets.  Or at least try (I guess this is gonna push back the release pretty hard but oh well, might as well do it now).

I *guess* a general good idea is to see what stuff like top has, and see if we can also display similar data.  Whether all of these will be available cross-platform is another story.

...also I might rebind the maximize key to ``M``.  Just gonna put that out there.  ``Enter`` is too useful of a key and makes sense for other tasks IMO.

# Possible candidates:

## CPU
Overall I think I might switch to heim purely for functionality reasons first, then go about implementing extra details.  I wouldn't graph most of these but just display them, maybe a Paragraph widget or something.

- [ ] Frequency (heim and sysinfo both support this)
- [ ] CTX-switch and interrupts (heim)
- [ ] Software-interrupts (linux only, heim)
- [ ] Load average
- [ ] Apparently someone is interested in CPI.  I _guess_ this is doable... maybe?  I don't know.
- [ ] At least for linux, heim _does_ support all cpu times.  I can display this (per cpu + avg) if that's desirable.

## RAM
- [ ] Could be interesting to display stuff like page faults?  The main problem is that neither heim nor sysinfo support this - and this might end up being a Linux-specific implementation for a while (macOS and Windows are ??? for me as of writing this).  Which I kinda hate.
- [ ] buff/cache values

## Processes
Some of these might just become columns overall but given a later priority in terms of drawing.

Also something unrelated is allowing disabling certain columns (some people probably won't care about this (and other extra columns)).
- [ ] Give an easy count to how many tasks, task state (sleep/zombie/running/stopped)
- [ ] Add process state.
- [ ] Showing actual values of mem usage?
- [ ] Virtual memory?
- [ ] I/O stats a la #58 (though that might just go there in general).  Doing this will close that task too.

## Network
- [ ] #52 might be a good fit here.  The smaller minimized version might be too chaotic with a small space right now (though that might change in the future if I can get rid of the table legend, and that's also a bit of a lie when you look at CPU graphs tbh).  Doing this will close that task.
- [ ] NIC data (see heim)
- [ ] May want to increase network to 10GB/s overall, or at least here.

## Temps
- [ ] As mentioned, graph this data.  Scale it to... I think 100 degrees is fine?  Maybe set it to like -10 to 110 C (dunno how many people would need subzero let's be real)

## Disks
- [ ] As mentioned, graph IO/s.  Probably range logarithmically from 0B/s to like... 10GB/s?  Maybe a bit higher?  Any more seems like overkill quite frankly.  Maybe 1TB/s?

**Comment by ClementTsang** [comment]
Note to self, probably going to break this up into multiple parts and potentially stretch it across multiple versions.  It's currently a bit too large in terms of scope as *one* issue IMO.

**Comment by ClementTsang** [comment]
Pushing to 0.4.0, going to split up too.

**Comment by kaushalyap** [comment]
@ClementTsang I cannot see memory legend in memory block, should I file a bug?

![Screenshot from 2020-04-02 12-14-08](https://user-images.githubusercontent.com/24698778/78218541-7c316e80-74ad-11ea-8db1-75f069a6da47.png)

**Comment by kaushalyap** [comment]
@ClementTsang  Why the memory legend not shown in the memory block? should I file a bug?

**Comment by ClementTsang** [comment]
> @ClementTsang Why the memory legend not shown in the memory block? should I file a bug?

This is due to the TUI library I'm using automatically hiding legends on certain heights.  This should be resolved when the library updates, as that will allow more controllable legends.

See: https://github.com/fdehau/tui-rs/issues/221

**Comment by tommywalkie** [comment]
@ClementTsang Recently tried Bottom and I'm enjoying it. 
I came here while looking for a way to display memory usage value instead of percentages. 
>  Showing actual values of mem usage?

Otherwise, nice work !

**Comment by ClementTsang** [comment]
Yeah, planning on adding that in (and eventually making stuff like the process widget more configurable in letting you choose what things you want to prioritize in seeing/see in general).

And thanks, glad you like it!
